libraries/AP_Scripting/examples/copter-fly-horizontal-circle.lua

```diff
@@ -0,0 +1,83 @@
+-- command a Copter to takeoff to 5m and fly a vertical circle in the clockwise direction
+--
+-- CAUTION: This script only works for Copter
+-- this script waits for the vehicle to be armed and RC6 input > 1800 and then:
+--    a) switches to Guided mode
+--    b) takeoff to 5m
+--    c) flies a vertical circle using the velocity controller
+--    d) switches to RTL mode
+
+
+local takeoff_alt_above_home = 5
+local copter_guided_mode_num = 4
+local copter_rtl_mode_num = 6
+local stage = 0
+local circle_angle = 0
+local circle_angle_increment = 1    -- increment the target angle by 1 deg every 0.1 sec (i.e. 10deg/sec)
+local circle_speed = 1              -- velocity is always 1m/s
+local yaw_cos = 0                   -- cosine of yaw at takeoff
+local yaw_sin = 0                   -- sine of yaw at takeoff
+
+-- the main update function that uses the takeoff and velocity controllers to fly a rough square pattern
+function update()
+  if not arming:is_armed() then -- reset state when disarmed
+    stage = 0
+    circle_angle = 0
+  else
+    pwm6 = rc:get_pwm(6)
+    if pwm6 and pwm6 > 1800 then    -- check if RC6 input has moved high
+      if (stage == 0) then          -- change to guided mode
+        if (vehicle:set_mode(copter_guided_mode_num)) then  -- change to Guided mode
+          local yaw_rad = ahrs:get_yaw()
+          yaw_cos = math.cos(yaw_rad)
+          yaw_sin = math.sin(yaw_rad)
+          stage = stage + 1
+        end
+      elseif (stage == 1) then      -- Stage1: takeoff
+        if (vehicle:start_takeoff(takeoff_alt_above_home)) then
+          stage = stage + 1
+        end
+      elseif (stage == 2) then      -- Stage2: check if vehicle has reached target altitude
+        local home = ahrs:get_home()
+        local curr_loc = ahrs:get_location()
```

Comment:
Easier to change the location alt frame to above home and read the alt directly.